call-a-function-api-oci-clouddev.md

```diff
@@ -0,0 +1,772 @@
+
+With your API Gateway and deployment created, you can now test you installation. Create a simple script for the `curl` command. To create the URL for `curl`, add your deployment path to your endpoint.
+
+1. Create the script file: `touch gtw01.sh && chmod 755 gtw01.sh`
+2. Add the command curl command to the script file:
+
+    ```console
+    #!/bin/bash
+    curl <your-api-gateway-endpoint>/http-info
+    ```
+
+3. The command returns: `{"message":"Hello World"}`
+
+You have connected your API Gateway to a boiler plate Python function. Next, you update your Python function to display information passed in an HTTP request.
+
+  
+## 4. Update Function to Access HTTP and Function Data
+
+Next, modify the boiler plate Python function to access the runtime context and display HTTP information.
+
+### Review Starting Python Code
+
+If you look at the boiler plate function, your Python function looks something like this.
+```console
+    import io
+    import json
+    import logging
+    
+    from fdk import response
+    
+    
+    def handler(ctx, data: io.BytesIO = None):
+        name = "World"
+        try:
+            body = json.loads(data.getvalue())
+            name = body.get("name")
+        except (Exception, ValueError) as ex:
+            logging.getLogger().info('error parsing json payload: ' + str(ex))
+        
+        logging.getLogger().info("Inside Python Hello World function")
+        return response.Response(
+           ctx, response_data=json.dumps(
+               {"message": "Hello {0}".format(name)}),
+           headers={"Content-Type": "application/json"}
+        )
+```
+
+Using this code as a starting point, the sections that follow convert the function into a Python function that returns HTTP and configuration data.
+
+### Update Required Packages
+
+First, update the function for required packages.
+
+1. Update the `requirements.txt` file for the `oci` package.
+
+    ```console
+        fdk
+        oci 
+    ```                   
+                    
+
+2. Update the `import` statements in `func.py` for required packages for the HTTP features: 
+
+    ```console
+        import io
+        import json
+        import oci
+        import logging
+        from urllib.parse import urlparse, parse_qs
+    ```                    
+
+    The `oci` package is required for some of the context requests. The `urlparse, parse_qs` packages are used for parsing.
+
+### Add HTTP Request Information
+
+First, remove the main body of the function. The `response` method and related code are added back as we go.
+
+```console
+    import io
+    import json
+    import oci
+    import logging
+    from urllib.parse import urlparse, parse_qs  
+                    
+    from fdk import response
+    
+    def handler(ctx, data: io.BytesIO = None): 
+```               
+
+Next add code to display HTTP information in the response. Here is the code with comments following.
+
+```console
+    import io
```

Comment:
outdent one level. First line of code should start at the indentation level of the fence (in this case left margin)